### PR TITLE
Week 3 ola

### PR DESCRIPTION
**Pull request by leealex88**
still working on  it.

### CODE REVIEW

**Comment by leealex88** [comment]
Homework week-3

File: week-3/Homework/mandatory/2-bush-berries.js [hunk 1]
```diff
@@ -10,8 +10,13 @@
   Use the tests to confirm which message to return
 */
 
-function bushChecker() {
-
+function bushChecker(arr) {
+  let allTheSame = arr.every(element => element === "pink")
+  if (allTheSame === true) {
```
Comment (naderakhgari) [inline week-3/Homework/mandatory/2-bush-berries.js]: you can use if statement in easy way like this:
if (allTheSame){
}

File: week-3/Homework/mandatory/5-journey-planner.js [hunk 2]
```diff
@@ -7,16 +7,21 @@
   NOTE: only the names should be returned, not the means of transport.
 */
 
-function journeyPlanner() {
-
+function journeyPlanner(arr, commuteType) {
+  return arr.filter(element => element.includes(commuteType)).map(element => element[0])
```
Comment (naderakhgari) [inline week-3/Homework/mandatory/5-journey-planner.js]: I like this type of coding, it looks quite perfect.

File: week-3/Homework/mandatory/7-password-validator.js [hunk 3]
```diff
@@ -23,47 +23,49 @@ PasswordValidationResult=  [false, false, false, false, true]
 */
 
 function validatePasswords(passwords) {
-
+  const check = passwords.filter(element => element.includes(element.length >= 5) && element === "A")
```
Comment (naderakhgari) [inline week-3/Homework/mandatory/7-password-validator.js]: to check the length of password you don't need to use of includes, and for check the other conditions you  need to use of regEx and test method

File: week-3/Homework/mandatory/1-oxygen-levels.js [hunk 4]
```diff
@@ -1,40 +1,42 @@
...
+
+// Their computer detects a list of nearby planets that have Oxygen in their atmosphere.
+
+// To be safe, they need to land on the first unamed planet that has Oxygen levels between 19.5% and 23.5%.
+
+// Write a function that finds the oxygen level of the first safe planet - Oxygen between 19.5% and 23.5%
+// */
+
+// function safeLevels(arr) {
+//   return arr.find(element => element > "19.5%" && element < "23.5%")
```
Comment (40thieves) [inline week-3/Homework/mandatory/1-oxygen-levels.js]: Looks like this has been commented out for some reason?

Your solution is really close, but I don't think this is quite right.

The problem here is that you're using the greater than and less than operators with strings not numbers.

Try changing one of the values in the test to "100.5%". Does it do what you expected? How could you fix it?

File: week-3/Homework/mandatory/7-password-validator.js [hunk 5]
```diff
@@ -23,47 +23,49 @@ PasswordValidationResult=  [false, false, false, false, true]
...
 }
 
 /* ======= TESTS - DO NOT MODIFY ===== */
 
 const passwords1 = ["Se%5", "TktE.TJTU", "384HsHF", "dvyyeyY!5", "tryT3729."]
-const passwords2 = ["StUFf27", "PlEnty", "Jai.33" "shajsaUA**&&", "PlEnty"]
+const passwords2 = ["StUFf27", "PlEnty", "Jai.33"
```
Comment (40thieves) [inline week-3/Homework/mandatory/7-password-validator.js]: It looks like you've accidentally removed a comma here which is causing a syntax error.

File: week-3/Homework/mandatory/2-bush-berries.js (hunk 1 above)
Comment (AlexeyBrambalov) [reply week-3/Homework/mandatory/2-bush-berries.js]: or you can do it all in one nice line 

 return  colours.every(colour => colour === "pink") ? "Bush is safe to eat from" : "Toxic! Leave bush alone!"

File: week-3/Homework/mandatory/4-eligible-students.js [hunk 6]
```diff
@@ -7,10 +7,37 @@
...
+//     if (element[1] >= 8) {
+//       namesMinimumEight.push(element[0])
+//     }
+//   })
+//   return namesMinimumEight
+// }
+
+function eligibleStudents(arr) {
+  const namesMinimumEight = []
+  arr.forEach(function (element) {
```
[the remark this answers is not in the document]
Comment (Sonjide) [reply week-3/Homework/mandatory/4-eligible-students.js]: great Ola

**Pull request closed without merging**